apollo-at-link/src/federation_schema/remove.rs

```diff
@@ -0,0 +1,2131 @@
+use crate::federation_schema::{OptionLinksMetadata, ReferencerFederationSchemaMut};
+use crate::location::{
+    DirectiveArgumentDefinitionLocation, DirectiveDefinitionLocation, EnumTypeDefinitionLocation,
+    EnumValueDefinitionLocation, InputObjectFieldDefinitionLocation,
+    InputObjectTypeDefinitionLocation, InterfaceFieldArgumentDefinitionLocation,
+    InterfaceFieldDefinitionLocation, InterfaceTypeDefinitionLocation,
+    ObjectFieldArgumentDefinitionLocation, ObjectFieldDefinitionLocation,
+    ObjectTypeDefinitionLocation, ScalarTypeDefinitionLocation, SchemaDefinitionLocation,
+    SchemaRootDefinitionKind, SchemaRootDefinitionLocation, UnionTypeDefinitionLocation,
+};
+use crate::referencer::{
+    DirectiveReferencers, EnumTypeReferencers, InputObjectTypeReferencers,
+    InterfaceTypeReferencers, ObjectTypeReferencers, Referencers, ScalarTypeReferencers,
+    UnionTypeReferencers,
+};
+use apollo_compiler::schema::{
+    Component, ComponentStr, Directive, EnumType, EnumValueDefinition, ExtendedType,
+    FieldDefinition, InputObjectType, InputValueDefinition, InterfaceType, Name, ObjectType,
+    ScalarType, UnionType,
+};
+use apollo_compiler::{Node, Schema};
+use indexmap::Equivalent;
+use std::hash::Hash;
+use std::ops::Deref;
+
+impl SchemaDefinitionLocation {
```

Comment:
I'd have this impl next to the actual definition of `SchemaDefinitionLocation`. We usually like to have impls along with struct definitions.

Reply:
This comment is for all of these `impl`s in this file ^